Plugins/MatrixPlugin/MatrixPlugin.cs

```diff
@@ -0,0 +1,149 @@
+using Chummer;
+using Chummer.Plugins;
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Threading.Tasks;
+using NLog;
+using Microsoft.ApplicationInsights.Channel;
+using System.Diagnostics;
+using System.Xml.XPath;
+using System.Xml.Serialization;
+using System.Xml;
+
+namespace MatrixPlugin
+{
+    [DebuggerDisplay("{" + nameof(GetDebuggerDisplay) + "(),nq}")]
+    public class MatrixPlugin : IPlugin
+    {
+
+        private static readonly Logger _log = LogManager.GetCurrentClassLogger();
+        public List<MatrixAction> ActionList;
+        public List<Program> ProgramList;
+
+        public override string ToString()
+        {
+            return "Matrix Helper";
+        }
+
+        public void CustomInitialize(frmChummerMain mainControl)
+        {
+            try
+            {
+                ActionList = new List<MatrixAction>();
+                XmlSerializer xmlSerializerAction = new XmlSerializer(typeof(MatrixAction));
+                XPathNavigator navActions = XmlManager.LoadXPath("actions.xml");
+                foreach (XPathNavigator xAction in navActions.Select("/chummer/actions/action"))
+                {
+                    XmlReader reader = xAction.ReadSubtree();
+                    MatrixAction item = (MatrixAction)xmlSerializerAction.Deserialize(reader);
+                    if (!string.IsNullOrEmpty(item.ActionSkill) && !string.IsNullOrEmpty(item.ActionAttribute))
+                        ActionList.Add(item);
+                }
+
+                ProgramList = new List<Program>();
+                XPathNavigator navSoftwares = XmlManager.LoadXPath("..\\Plugins\\MatrixPlugin\\data\\program.xml");
```

Comment:
... this is not what I was expecting. I mean, I guess it works right now, but it really is not the intended behavior of `XmlManager.LoadXPath`, which should not be able to move out of the `data` directory as a baseline at all. Otherwise, it can be a potential security risk when it comes time to processing custom files. Do not expect this to keep working this way in the future, because this *is* a potential security risk and I *will* patch it up.